femtools/Time_Period.F90

```diff
@@ -0,0 +1,182 @@
+! Handles output time period logic
+
+#include "fdebug.h"
+
+module time_period
+  use FLDebug
+  use global_parameters, only: OPTION_PATH_LEN
+  use embed_python
+  use spud
+  use parallel_tools
+  use timers
+
+  implicit none
+
+  type time_period_type
+    private
+    
+    logical :: last_times_initialised = .false.
+    
+    real :: real_dump_period = 0.0
+    integer :: int_dump_period = 0    
+    real :: last_dump_time
+    
+    real :: last_dump_cpu_time
+    real :: last_dump_wall_time
+  end type time_period_type
+
+contains
+  function should_output(CS, current_time, timestep, option_path)
```

Comment:
Could you add a short description add the top of each function/subroutine in this module? And clarify arguments where necessary?